### PR TITLE
Ability to choose which method to use when generating slugs

### PR DESCRIPTION
**Pull request by mehranrasulian**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -96,7 +96,7 @@ Next up, the service provider must be registered:
 
 You can publish the migration with:
 ```bash
-php artisan vendor:publish --provider="Spatie\Tags\TagsServiceProvider" --tag="migrations"
+php artisan vendor:publish --provider="Spatie\Tags\TagsServiceProvider"
```
Comment (freekmurze) [inline README.md]: This change is not related to the rest of the PR, please remove this change.

File: config/tags.php [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<?php
+
+return [
+    // The given function generates a URL friendly "slug" from the tag name property before saving it.
+    'normalizer' => 'str_slug',
```
Comment (freekmurze) [inline config/tags.php]: The naming could be improved. What do you think of `slugger`?

File: src/TagsServiceProvider.php [hunk 3]
```diff
@@ -18,6 +18,10 @@ public function boot()
                     __DIR__.'/../database/migrations/create_tag_tables.php.stub' => database_path('migrations/'.$timestamp.'_create_tag_tables.php'),
                 ], 'migrations');
             }
+
+            $this->publishes([
+                __DIR__.'/../config/tags.php' => config_path('tags.php'),
```
Comment (freekmurze) [inline src/TagsServiceProvider.php]: When naming the config file, use the entire package name `laravel-tags.php`

**Comment by mehranrasulian** [comment]
I didn't see any need to add the tests for this feature. Is there any need ?

File: config/tags.php (hunk 2 above)
Comment (mehranrasulian) [inline config/tags.php]: Yes, slugger is better :)

File: src/TagsServiceProvider.php (hunk 3 above)
Comment (mehranrasulian) [inline src/TagsServiceProvider.php]: Sure

**Comment by freekmurze** [comment]
> I didn't see any need to add the tests for this feature. Is there any need ?

Yeah I want to make 100 % sure that the added functionality works. Also make sure that the package continues to work properly when no slugger has been set (backwards compatibility)

**Comment by mehranrasulian** [comment]
Sorry, I'm still not good at naming test functions. do the names make sense?

File: tests/TagTest.php [hunk 4]
```diff
@@ -43,6 +43,26 @@ public function it_automatically_generates_a_slug()
...
+    {
+        app('config')->set('laravel-tags.slugger', '');
+
+        $tag = Tag::findOrCreateFromString('this is a tag');
+
+        $this->assertSame('this-is-a-tag', $tag->slug);
+    }
+
+    /** @test */
+    public function it_uses_the_specified_method_defiend_in_config_slugger_to_generate_a_slug()
```
Comment (freekmurze) [inline tests/TagTest.php]: Maybe `it_can_use_a_custom_slugger` is better here.

File: tests/TagTest.php [hunk 5]
```diff
@@ -43,6 +43,26 @@ public function it_automatically_generates_a_slug()
...
+    {
+        app('config')->set('laravel-tags.slugger', 'str_slug');
```
Comment (freekmurze) [inline tests/TagTest.php]: Use another function that produces a different result than the `str_slug` function. Perhaps `strtoupper`? In that way we know for certain that the custom function is being used.

**Comment by freekmurze** [comment]
Thank you!

**Comment by mehranrasulian** [comment]
You are welcome!
